robot/web/web_c.go

```diff
@@ -84,150 +82,83 @@ func (svc *webService) streamInitialized() bool {
+		// already exists.
+		//
+		// TODO(RSDK-9079) Add reliable framerate fetcher for stream videosources
+		stream, err := svc.createStream(config, name)
 		if err != nil {
 			return err
-		} else if alreadyRegistered {
-			continue
 		}
 
-		svc.startVideoStream(ctx, source, stream)
+		// If the stream already exists, `createStream` will return a `nil` stream and `nil`
+		// error. `startVideoStream` copes with this input.
+		svc.startVideoStream(ctx, svc.videoSources[name], stream)
 	}
-
-	for name, source := range svc.audioSources {
-		const isVideo = false
-		stream, alreadyRegistered, err := newStream(name, isVideo)
+	for name := range svc.audioSources {
+		// Similarly, we walk the updated set of `audioSources` and ensure all of the audio sources
+		// are "created" and "started". `createStream` and `startAudioStream` have the same
+		// behaviors as described above for video streams.
+		config := gostream.StreamConfig{
+			Name:                name,
+			AudioEncoderFactory: svc.opts.streamConfig.AudioEncoderFactory,
+		}
+		stream, err := svc.createStream(config, name)
 		if err != nil {
 			return err
-		} else if alreadyRegistered {
-			continue
 		}
-
-		svc.startAudioStream(ctx, source, stream)
+		svc.startAudioStream(ctx, svc.audioSources[name], stream)
 	}
-
 	return nil
 }
 
-func (svc *webService) makeStreamServer(ctx context.Context) (*StreamServer, error) {
-	svc.refreshVideoSources()
-	svc.refreshAudioSources()
-	var streams []gostream.Stream
-	var streamTypes []bool
-
-	if svc.opts.streamConfig == nil || (len(svc.videoSources) == 0 && len(svc.audioSources) == 0) {
-		if len(svc.videoSources) != 0 || len(svc.audioSources) != 0 {
-			svc.logger.Debug("not starting streams due to no stream config being set")
-		}
-		noopServer, err := webstream.NewServer(streams, svc.r, svc.logger.Sublogger("stream"))
-		return &StreamServer{noopServer, false}, err
-	}
-
-	addStream := func(streams []gostream.Stream, name string, isVideo bool) ([]gostream.Stream, error) {
-		config := gostream.StreamConfig{
-			Name: name,
-		}
-		if isVideo {
-			config.VideoEncoderFactory = svc.opts.streamConfig.VideoEncoderFactory
-
-			// set TargetFrameRate to the framerate of the video source if available
-			props, err := svc.videoSources[name].MediaProperties(ctx)
-			if err != nil {
-				svc.logger.Warnw("failed to get video source properties", "name", name, "error", err)
-			} else if props.FrameRate > 0.0 {
-				// round float up to nearest int
-				config.TargetFrameRate = int(math.Ceil(float64(props.FrameRate)))
-			}
-			// default to 60fps if the video source doesn't have a framerate
-			if config.TargetFrameRate == 0 {
-				config.TargetFrameRate = 60
-			}
-
-			if runtime.GOOS == "windows" {
-				// TODO(RSDK-1771): support video on windows
-				svc.logger.Warnw("not starting video stream since not supported on Windows yet", "name", name)
-				return streams, nil
```

Comment:
hmmm. Do we know that we can run streams on windows now?

Reply:
Good call - will add Windows check back in.

Reply:
Added in windows check for video streams.
Q: should we also skip audio?

Reply:
[this](https://github.com/viamrobotics/rdk/blob/main/web/server/entrypoint_windows.go) seems to imply that opus runs fine on windows.

Reply:
disclaimer I do not know if the implication is the truth.

Reply:
the static build file /etc/ld_wrappers.sh also implies that there is no special casing for opus, it includes a statically linked -lx264 but no opus in the static build flags.

Reply:
Nice find! 
Hm,  yeah I would be surprised if audio drivers worked inside WSL but we can go with it for now.